# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/components/sessionstore/test/browser_restore_session_in_undoCloseTab.js
# Commit: d29c1fad7862
# Full Hash: d29c1fad7862d61ef7c66d5989caa5ed0af4f011
# Author: Mike Conley <mconley@mozilla.com>
# Date: 2023-03-03 21:37:37
# Regressor Bug: 1423125
# File Overlap Count: 1
# Description:
#   Bug 1423125 - Restore previous session if undoCloseTab is called with no more tabs to restore. r=dao
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171377
# ==============================================================================

diff -r c81bb857ccff -r d29c1fad7862 browser/components/sessionstore/test/browser_restore_session_in_undoCloseTab.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/browser/components/sessionstore/test/browser_restore_session_in_undoCloseTab.js	Fri Mar 03 14:06:43 2023 +0000
@@ -0,0 +1,56 @@
+/* Any copyright is dedicated to the Public Domain.
+   http://creativecommons.org/publicdomain/zero/1.0/ */
+
+"use strict";
+
+const { _LastSession } = ChromeUtils.importESModule(
+  "resource:///modules/sessionstore/SessionStore.sys.mjs"
+);
+
+/**
+ * Tests that if the user invokes undoCloseTab in a window for which there are no
+ * tabs to undo closing, that we attempt to restore the previous session if one
+ * exists.
+ */
+add_task(async function test_restore_session_in_undoCloseTab() {
+  forgetClosedTabs(window);
+  registerCleanupFunction(() => {
+    forgetClosedTabs(window);
+  });
+
+  const state = {
+    windows: [
+      {
+        tabs: [
+          {
+            entries: [
+              {
+                url: "https://example.org/",
+                triggeringPrincipal_base64,
+              },
+            ],
+          },
+          {
+            entries: [
+              {
+                url: "https://example.com/",
+                triggeringPrincipal_base64,
+              },
+            ],
+          },
+        ],
+        selected: 2,
+      },
+    ],
+  };
+
+  _LastSession.setState(state);
+
+  let sessionRestored = promiseSessionStoreLoads(2 /* total restored tabs */);
+  let result = undoCloseTab();
+  await sessionRestored;
+  Assert.equal(result, null);
+  Assert.equal(gBrowser.tabs.length, 2);
+
+  gBrowser.removeAllTabsBut(gBrowser.tabs[0]);
+});